orchestrator/test_runner/src/transaction_stress_test.rs

```diff
@@ -88,41 +87,44 @@ pub async fn transaction_stress_test(
         );
     }
 
-    let start = Instant::now();
     let mut good = true;
-    while Instant::now() - start < TOTAL_TIMEOUT {
+    match tokio::time::timeout(TOTAL_TIMEOUT, async {
         good = true;
-        for keys in user_keys.iter() {
-            let c_addr = keys.cosmos_address;
-            let balances = contact.get_balances(c_addr).await.unwrap();
-            for token in erc20_addresses.iter() {
-                let mut found = false;
-                for balance in balances.iter() {
-                    if balance.denom.contains(&token.to_string())
-                        && balance.amount == one_hundred_eth()
-                    {
-                        found = true;
+        loop {
+            for keys in user_keys.iter() {
+                let c_addr = keys.cosmos_address;
+                let balances = contact.get_balances(c_addr).await.unwrap();
+                for token in erc20_addresses.iter() {
+                    let mut found = false;
+                    for balance in balances.iter() {
+                        if balance.denom.contains(&token.to_string())
+                            && balance.amount == one_hundred_eth()
+                        {
+                            found = true;
+                        }
+                    }
+                    if !found {
+                        good = false;
                     }
                 }
-                if !found {
-                    good = false;
-                }
+                tokio::time::sleep(Duration::from_secs(5)).await;
```

Comment:
We still need to include the `break` condition here, otherwise the loop will continue to run until the timeout is hit. The original issue was just that it was breaking out of the `for` loop and not the main `loop`. It would be here at the same scope just before the `sleep` call.

Reply:
I'm not too sure if this is what you mean, HMU if it's not what you want
https://github.com/PeggyJV/gravity-bridge/pull/274/commits/268d99579ae36e9856b34f75d5af6db7ea8705ac